cmd/kubeadm/app/cmd/alpha/kubelet.go

```diff
@@ -109,16 +111,39 @@ func newCmdKubeletConfigDownload() *cobra.Command {
 			client, err := kubeconfigutil.ClientSetFromFile(kubeConfigFile)
 			kubeadmutil.CheckErr(err)
 
-			err = kubeletphase.DownloadConfig(client, kubeletVersion, constants.KubeletRunDirectory)
+			criSocket, err := getCRISocket(criSocketStr)
+			if err != nil {
+				kubeadmutil.CheckErr(err)
+			}
+
+			nodeReg := kubeadmapi.NodeRegistrationOptions{
+				CRISocket: criSocket,
+			}
+
+			err = kubeletphase.DownloadConfig(client, kubeletVersion, &nodeReg, constants.KubeletRunDirectory)
 			kubeadmutil.CheckErr(err)
 		},
 	}
 
 	options.AddKubeConfigFlag(cmd.Flags(), &kubeConfigFile)
+	cmdutil.AddCRISocketFlag(cmd.Flags(), &criSocketStr)
 	cmd.Flags().StringVar(&kubeletVersionStr, "kubelet-version", kubeletVersionStr, "The desired version for the kubelet. Defaults to being autodetected from 'kubelet --version'.")
 	return cmd
 }
 
+func getCRISocket(criSocketStr string) (string, error) {
+	if len(criSocketStr) > 0 {
```

Comment:
This condition will always be false as this function is always called with an empty string. I agree with @kad here - we don't need this function.

Reply:
Please read line 129:

> Please read line 129: